### DISCUSSION TITLE
Size of a group

### DISCUSSION THREAD

**Discussion by MaestroKongrio** [semaphore-protocol/semaphore#286] [General]
Hi, I'm trying to buiild something like a "ZK National Identity Proof". My idea is let a national authority to build a semaphore group and make this information available on chain.
My concern is about max group size. Let's say we need to add 100 million people to a group. Is this doable? it's even possible to handle this kind of size?
Thanks,

**Comment by vplasencia** [comment]
Hi @MaestroKongrio! The max size is 2 ^ tree depth (https://semaphore.appliedzkp.org/docs/guides/groups#create-a-group) and the tree depth should be >=16 and <=32 (https://github.com/semaphore-protocol/semaphore/blob/main/packages/group/src/group.ts#L18)
